src/baguetteBox.js

```diff
@@ -172,18 +172,24 @@
             if (userOptions && userOptions.filter) {
                 regex = userOptions.filter;
             }
+            
+            // Get nodes from gallery elements or single-element galleries
+            var tagsNodeList = [];
+            if (galleryElement.tagName === 'A') {
+                tagsNodeList = [galleryElement];
+            } else {
+                tagsNodeList = galleryElement.getElementsByTagName('a');
+            }
+            
             // Filter 'a' elements from those not linking to images
-            var tagsNodeList = galleryElement.getElementsByTagName('a');
             tagsNodeList = [].filter.call(tagsNodeList, function(element) {
                 return regex.test(element.href);
             });
-            var gallery = [];
-
             if (tagsNodeList.length === 0) {
                 return;
             }
-            selectorData.galleries.push(gallery);
-
+            
+            var gallery = [];
```

Comment:
@feimosi @XhmikosR The order here didn't look right to me, IMHO. I re-ordered the gallery declaration and moved the `push` call after the `forEach` below. Hope this is right.